mainwindow.cpp

```diff
@@ -826,6 +850,13 @@ void MainWindow::updateSelectedObjects() {
             widget->setLayout(fl);
             frame->layout()->addWidget(widget);
 
+            // generate "Open in Text Editor" button
+            if (key == "script_label") {
+                QPushButton *scriptButton = new QPushButton(QString("Open This Map's Scripts in a Text Editor"));

```

Comment:
I don't think this button should be added in the context of the event.  If you have multiple events selected, you'll see this button for each selected event.  Instead, the button should probably just go underneath the "New Object" and "Delete" buttons.  Also, it's a bit too long.  Maybe just "Open Map Scripts"

Reply:
Oh I didn't realize you could have multiple selected events. Agree with both the comments-- will make appropriate changes. Did the button work for you though?

Reply:
Fixed